packages/file_selector/file_selector_windows/lib/src/dart_file_selector_api.dart

```diff
@@ -0,0 +1,328 @@
+        confirmButtonText: confirmButtonText,
+        selectionOptions: selectionOptions);
+  }
+
+  /// Returns a full path, including file name and it's extension, from user selection.
+  String? getSavePath({
+    String? initialDirectory,
+    String? confirmButtonText,
+    String? suggestedFileName,
+    SelectionOptions? selectionOptions,
+  }) {
+    fileMustExist = false;
+    final SelectionOptions defaultSelectionOptions = SelectionOptions(
+        allowMultiple: false, selectFolders: true, allowedTypes: <TypeGroup>[]);
+    return _getDirectory(
+        initialDirectory: initialDirectory,
+        confirmButtonText: confirmButtonText,
+        suggestedFileName: suggestedFileName,
+        selectionOptions: selectionOptions ?? defaultSelectionOptions);
+  }
+
+  /// Returns a list of file paths.
+  List<String> getFiles(
+      {String? initialDirectory,
+      String? confirmButtonText,
+      required SelectionOptions selectionOptions}) {
+    fileMustExist = false;
+    int hResult = initializeComLibrary();
+    final IFileOpenDialog fileDialog = FileOpenDialog.createInstance();
+    using((Arena arena) {
+      final Pointer<Uint32> ptrOptions = arena<Uint32>();
+
+      hResult = getOptions(ptrOptions, fileDialog);
+      hResult = setDialogOptions(ptrOptions, selectionOptions, fileDialog);
+    });
+
+    hResult = setInitialDirectory(initialDirectory, fileDialog);
+    hResult = addFileFilters(selectionOptions, fileDialog);
+    hResult = addConfirmButtonLabel(confirmButtonText, fileDialog);
+    hResult = _fileOpenDialogAPI.show(hWndOwner, fileDialog);
+
+    return returnSelectedElements(hResult, selectionOptions, fileDialog);
+  }
+
+  /// Returns dialog options.
+  @visibleForTesting
+  int getOptions(Pointer<Uint32> ptrOptions, IFileOpenDialog dialog) {
+    final int hResult = _fileOpenDialogAPI.getOptions(ptrOptions, dialog);
+    _validateResult(hResult);
+
+    return hResult;
+  }
+
+  /// Returns the dialog option based on conditions.
+  @visibleForTesting
+  int getDialogOptions(int options, SelectionOptions selectionOptions) {
+    if (!fileMustExist) {
+      options &= ~FILEOPENDIALOGOPTIONS.FOS_PATHMUSTEXIST;
+      options &= ~FILEOPENDIALOGOPTIONS.FOS_FILEMUSTEXIST;
+    }
+
+    if (selectionOptions.selectFolders) {
+      options |= FILEOPENDIALOGOPTIONS.FOS_PICKFOLDERS;
+    }
+
+    if (selectionOptions.allowMultiple) {
+      options |= FILEOPENDIALOGOPTIONS.FOS_ALLOWMULTISELECT;
+    }
+
+    return options;
+  }
+
+  /// Sets and checks options for the dialog.
+  @visibleForTesting
+  int setDialogOptions(Pointer<Uint32> ptrOptions,
+      SelectionOptions selectionOptions, IFileOpenDialog dialog) {
+    final int options = getDialogOptions(ptrOptions.value, selectionOptions);
+
+    final int hResult = _fileOpenDialogAPI.setOptions(options, dialog);
+
+    _validateResult(hResult);
+
+    return hResult;
+  }
+
+  /// Sets the initial directory to open the dialog
```

Comment:
Nit: end with a period.

```suggestion
  /// Sets the initial directory to open the dialog.
```